### PR TITLE
polkadot v0.9.18

### PR DESCRIPTION
**Pull request by zqhxuyuan**
companion:
- [x] **orml**: https://github.com/open-web3-stack/open-runtime-module-library/pull/718
- [x] **xcm-emulator**: https://github.com/shaunxw/xcm-simulator/pull/26
- [x] stable-asset: https://github.com/nutsfinance/stable-asset/pull/19
- [x] evm-tests: https://github.com/AcalaNetwork/evm-tests/pull/3 https://github.com/AcalaNetwork/evm-tests/pull/6

dep upgrade:
- [x] scale-info to 2.0.1
- [x] codec to 3.0.0
- [x] primitive-types to 0.11
- [x] upgrade evm related: ethereum-types,evm,ethereum, ~~current use our own fork: https://github.com/AcalaNetwork/evm/pull/2~~

changes:
- [x] remove ethers, use our own util method to generate evm address because of primitive-types conflict cause compile error
- [x] support relaychain rpc: https://github.com/paritytech/cumulus/pull/963
- [x] Statemine [IdentityFee](https://github.com/paritytech/cumulus/blob/polkadot-v0.9.17/polkadot-parachains/statemine/src/xcm_config.rs#L165)(4,000,000,000) to [WeightToFee](https://github.com/paritytech/cumulus/blob/polkadot-v0.9.18/polkadot-parachains/statemine/src/xcm_config.rs#L167)(10,666,664): https://github.com/paritytech/cumulus/pull/1055
  - fix test
  - ~~migration statemine fee~~
- [x] switch to wasm only 

issues: 
- closes https://github.com/AcalaNetwork/Acala/issues/2025 
- closes https://github.com/AcalaNetwork/Acala/issues/1961

### CODE REVIEW

**Comment by zqhxuyuan** [comment]
@ermalkaleci can you check why `make test-eth` failed? I run it on local is succeeded.

**Comment by ermalkaleci** [comment]
@zqhxuyuan failing after this merge commit
run cargo clean first to reproduce

**Comment by ermalkaleci** [comment]
> @zqhxuyuan failing after this merge commit
> run cargo clean first to reproduce

I think it's cargo cache issue. will clean and try again

**Comment by zjb0807** [comment]
I removed the `/home/acala/acala/actions-runner/_work/Acala/Acala/evm-tests/Cargo.lock` on the `gh-runner-1`. It shoud work now.

**Review by xlc — DISMISSED** [review]

File: node/service/src/chain_spec/mandala.rs [hunk 1]
```diff
@@ -45,25 +54,42 @@ pub fn parachain_dev_testnet_config(mnemonic: Option<&str>) -> Result<ChainSpec,
...
-		evm_accounts.push(wallet.address());
-		log::info!(
-			"index: {:?}, private_key: {:x?} address: {:?}",
-			index,
-			hex::encode(wallet.signer().to_bytes()),
-			wallet.address()
-		);
+		let addr = generate_evm_address::<English>(phrase, index);
+		evm_accounts.push(addr);
+		log::info!("index: {:?}, evm address: {:?}", index, addr);
```
Comment (zjb0807) [inline node/service/src/chain_spec/mandala.rs:92]: I don't know why, the log can't be printed here

**Review by zjb0807 — APPROVED** [review]
LGTM

File: node/service/src/chain_spec/mandala.rs (hunk 1 above)
Comment (zqhxuyuan) [reply node/service/src/chain_spec/mandala.rs:92]: e2e test can print: https://github.com/AcalaNetwork/Acala/runs/5827227801?check_suite_focus=true
image

I've checked that old ether-rs result is the same compare to our current implementation, so it wouldn't break anything.

**Review by ermalkaleci — APPROVED** [review]

**Review by ermalkaleci — APPROVED** [review]

**Pull request merged**